alternator/expression_factory.cc

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include "expressions.hh"
+#include "utils/log.hh"
+
+#include <boost/intrusive/list.hpp>
+#include <absl/container/node_hash_map.h>
+#include <optional>
+
+logging::logger cache_logger("alternator-cache");
```

Comment:
nitpick: please make the logger's name (the "alternator-cache") one related to the file's name if it's a file-local logger.

Also I suggest defining it as **static** and calling the variable **logger** - it makes everything easier to guess when you want to write logging code and need to start guessing the variable name.